packages/dd-trace/src/datastreams/writer.js

```diff
@@ -0,0 +1,66 @@
+const pkg = require('../../../../package.json')
+const log = require('../log')
+const request = require('../exporters/common/request')
+const { URL, format } = require('url')
+const msgpack = require('msgpack-lite')
+const zlib = require('zlib')
+const codec = msgpack.createCodec({ int64: true })
+
+function makeRequest (data, url, cb) {
```

Comment:
Why is an entire new `makeRequest` required? Why can't the existing code be reused?

Reply:
We already have two versions and this new version is different enough from the existing ones that I don't think we benefit much from code reuse.